script/Deploy.s.sol

```diff
@@ -10,12 +10,40 @@ import {ERC6538Registry} from "src/ERC6538Registry.sol";
 contract Deploy is Script {
   ERC5564Announcer announcer;
   ERC6538Registry registry;
+  address deployer = 0x4e59b44847b379578588920cA78FbF26c0B4956C;
+  bytes32 salt = "";
 
   function run() public {
+    bytes memory ERC5564CreationCode = abi.encodePacked(type(ERC5564Announcer).creationCode);
+    bytes memory ERC6538CreationCode = abi.encodePacked(type(ERC6538Registry).creationCode);
+    address ERC5564ComputedAddress = computeAddress(salt, keccak256(ERC5564CreationCode), deployer);
+    address ERC6538ComputedAddress = computeAddress(salt, keccak256(ERC6538CreationCode), deployer);
+
     vm.broadcast();
-    announcer = new ERC5564Announcer();
+    announcer = new ERC5564Announcer{salt: salt}();
 
     vm.broadcast();
-    registry = new ERC6538Registry();
+    registry = new ERC6538Registry{salt: salt}();
+
+    require(address(announcer) == ERC5564ComputedAddress);
+    require(address(registry) == ERC6538ComputedAddress);
+  }
+
+  function computeAddress(bytes32 _salt, bytes32 _bytecodeHash, address _deployer)
```

Comment:
forge-std has this method which you can use instead of re-implementing it

Reply:
Thanks! Now using `computeCreate2Address` in https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5/commits/4460b6edabe9487c5157a75f918ee7c116919931